Clone.xs

```diff
@@ -290,31 +290,54 @@ sv_clone (SV * ref, HV* hseen, int depth)
               obj = mg->mg_obj; 
               break;
             case 't':	/* PERL_MAGIC_taint */
-	      continue;
-              break;
-            case '<':	/* PERL_MAGIC_backref */
-	      continue;
-              break;
+            case '<': /* PERL_MAGIC_backref */
             case '@':  /* PERL_MAGIC_arylen_p */
-             continue;
+              continue;
               break;
             case 'P': /* PERL_MAGIC_tied */
             case 'p': /* PERL_MAGIC_tiedelem */
             case 'q': /* PERL_MAGIC_tiedscalar */
-	      magic_ref++;
+	            magic_ref++;
 	      /* fall through */
             default:
               obj = sv_clone(mg->mg_obj, hseen, -1); 
           }
         } else {
           TRACEME(("magic object for type %c in NULL\n", mg->mg_type));
         }
-	/* this is plain old magic, so do the same thing */
-        sv_magic(clone, 
-                 obj,
-                 mg->mg_type, 
-                 mg->mg_ptr, 
-                 mg->mg_len);
+
+        { /* clone the mg_ptr pv */
+          char *mg_ptr = mg->mg_ptr; /* default */
+
+          if (mg->mg_len >= 0) { /* copy the pv */
+            if (mg_ptr) {
+              Newxz(mg_ptr, mg->mg_len+1, char); /* add +1 for the NULL at the end? */
+              Copy(mg->mg_ptr, mg_ptr, mg->mg_len, char);
+            }
+          } else if (mg->mg_len == HEf_SVKEY) {
+            /* let's share the SV for now */
+            SvREFCNT_inc((SV*)mg->mg_ptr);
+            /* maybe we also want to clone the SV... */
+            //if (mg_ptr) mg->mg_ptr = (char*) sv_clone((SV*)mg->mg_ptr, hseen, -1); 
```

Comment:
maybe this is what we should prefer doing, the current solution above just do one SvRefCnt++